src/sparql.rs

```diff
@@ -82,8 +82,17 @@ impl<'a> QueryableDataset<'a> for &'a Hdt {
     }
 }
 
-pub fn query<'a>(q: &str, hdt: &'a Hdt) -> Result<spareval::QueryResults<'a>, QueryEvaluationError> {
-    let query = SparqlParser::new().parse_query(q).unwrap_or_else(|_| panic!("error processing query {q}"));
+pub fn query<'a>(
+    q: &str, hdt: &'a Hdt, base_iri: Option<&str>,
```

Comment:
@KonradHoeffner what do you think about this? Relates to https://github.com/KonradHoeffner/hdt/pull/66#issuecomment-3135133402

Per the w3c spec, rdf and sparql files with an empty base iri are valid, so I'm not sure it makes sense to panic here if it's missing. Does the `Option` work here? Should there be a `query_with_base_iri` function?